### DISCUSSION TITLE
Any way to use node-fetch on server api folder instead of undici?

### DISCUSSION THREAD

**Discussion by ptdev** [nuxt/nuxt#26246] [Questions]
Hi,

I came across a strange issue while developing a Nuxt app that I've pinpointed the cause as (almost for sure) being some issue with node's default fetch since a simple proof of concept shows it working with node-fetch but not with native fetch.

I have submitted an[ issue on node's help repo](https://github.com/nodejs/help/issues/4357) in case you want more details or try the proof of concept on your end (appreciated).

Now my issue with nuxt, while trying to fix that issue would be to just use node-fetch on my /server folder endpoints and be done with it.

But when I add `import fetch from 'node-fetch'` at the top of a /server/api file I would assume that any further fetch requests inside that file would use node-fetch and not the native fetch, but apparently, even with the import, the error I'm getting is from undici (which I believe is node's default fetch).

So my question is: Is there any way to use node-fetch on server endpoints? 

Apologies if I'm missing something obvious.

Thanks, cheers.

**Comment by topd5** [comment]
Have you tried to rename the import?
```
import nodeFetch from 'node-fetch';
```

**Reply by ptdev** [reply]
If I place that on top of a server api file, and then do a request using nodeFetch, the error I get is from undici.

But only if I call that endpoint from my nuxt app.

If I create a proof of concept file that uses just node directly, then it works correctly.

It's as if Nuxt is somehow overriding all fetch requests inside the defineEventHandler  (?)
